server/src/utils/callAutomationUtils.ts

```diff
@@ -0,0 +1,294 @@
+    CALLCONNECTION_ID_TO_CORRELATION_ID[key].serverCallId.includes(serverCallId)
+  );
+
+  /**
+   * Check if the call automation client and connection for the call already exists. If it does, we don't need to create a new one.
+   */
+  if (connectionId) {
+    console.log('Call connection for call already exists:', connectionId);
+    return;
+  }
+
+  const options = {
+    callIntelligenceOptions: {
+      cognitiveServicesEndpoint: callAutomationConfig?.CognitionAPIEndpoint ?? ''
+    },
+    transcriptionOptions: transcriptionOptions
+  };
+  const res = await getCallAutomationClient().connectCall(
+    {
+      kind: 'serverCallLocator',
+      id: serverCallId
+    },
+    callAutomationConfig?.ServerHttpUrl ?? '',
+    options
+  );
+  console.log('Connect call result', res);
+  const callConnection = res.callConnection;
+  console.log('Call connection', callConnection);
+};
+
+export const startTranscriptionForCall = async (
+  callConnectionId: string,
+  options?: TranscriptionOptions
+): Promise<void> => {
+  console.log('Starting transcription for call:', callConnectionId);
+  console.log(CALLCONNECTION_ID_TO_CORRELATION_ID);
+  const callConnection = await getCallAutomationClient().getCallConnection(callConnectionId);
+
+  return await callConnection.getCallMedia().startTranscription(options);
+};
+
+export const stopTranscriptionForCall = async (callConnectionId: string): Promise<void> => {
+  console.log('Stopping transcription for call:', callConnectionId);
+  console.log(CALLCONNECTION_ID_TO_CORRELATION_ID);
+  const callConnection = await getCallAutomationClient().getCallConnection(callConnectionId);
+  return await callConnection.getCallMedia().stopTranscription();
+};
+
+/**
+ * We should be storing this based on the connectionId and not the correlationId
+ * We should re-write all the fetches based on the callId to use the serverCallId. This is
+ * because all of the clients that join can have different callId's and we need to make sure we are
+ * pulling the correct transcription data.
+ */
+export const TRANSCRIPTION_STORE: { [key: string]: Partial<CallTranscription> } = {};
+/**
+ * Used to map between the call connection id and the correlation id from both transcription and
+ * call automation events.
+ */
+export const CALLCONNECTION_ID_TO_CORRELATION_ID: {
+  [key: string]: { correlationId?: string; serverCallId: string };
+} = {};
+
+/**
+ * used to store the remote participants in the call
+ * This object is keyed off the callId and contains the communicationUserId and displayName of the remote participants
+ *
+ * Keeps track of all participants who have ever joined the call so we can show their display name in the transcription and summary.
+ */
+export const REMOTE_PARTICIPANTS_IN_CALL: {
+  [key: string]: { communicationUserId: string; displayName: string }[];
+} = {};
+
+/**
+ * used to store the local participant in the call
```

Comment:
Does this only support a single local participant? What if two different calls are happen simultaneously?

Reply:
this is a good point we should be processing all the participants together. this is a work in progress for adding the names to the downloaded transcript so I will remove this and update it so that we are just tracking all the participants together.